### ISSUE TITLE
Highlight active toolbar buttons on hover on macOS

### ISSUE THREAD

**Issue by cbrnr** [cbrnr/mnelab#460]
On macOS, Qt doesn't highlight [toolbar buttons](https://doc.qt.io/qt-6/qtoolbutton.html#details) on hover (they call it "autoraise"), see [here](https://doc.qt.io/qt-6/qtoolbutton.html#autoRaise-prop). This is quite unfortunate, because (1) native macOS application do highlight on hover, and (2) from a usability point of view, it is really difficult to discover if a button is active or not.

Although this is not an MNELAB issue, maybe we could implement a workaround as outlined [here](https://stackoverflow.com/a/11852107/1112283) if it is not too hacky (e.g., it needs to support both light and dark themes).

**Comment by hoechenberger** [comment]
One of the challenges I foresee for the workaround implementation is properly dealing with the OS theme (light vs dark)

**Comment by cbrnr** [comment]
Yes, maybe, although it is possible to determine which system theme is active (and there is also a signal which is emitted if the theme changes).

For reference, here's the toolbar of a native macOS app (Notes) in both light and dark modes:


https://github.com/user-attachments/assets/a7a22339-e442-42de-b4f4-81e596cd78d8


https://github.com/user-attachments/assets/d75453dd-7b83-440f-a8ee-e92904dd8ec2
